validator/src/commands/run/args.rs

```diff
@@ -1190,6 +1190,40 @@ pub fn add_args<'a>(app: App<'a, 'a>, default_args: &'a DefaultArgs) -> App<'a,
                  accounts index at the cost of index performance.",
             ),
     )
+    .arg(
+        Arg::with_name("accounts_index_limit_entries")
+            .long("accounts-index-limit-entries")
+            .value_name("ENTRIES")
+            .takes_value(true)
+            .conflicts_with("enable_accounts_disk_index")
+            .conflicts_with("accounts_index_limit_mb")
+            .help("Flush accounts index to disk when entry count exceeds this limit")
+            .long_help(
+                "Sets an entry count threshold for the in-memory accounts index. When the total \
+                 number of entries in the index exceeds this limit, it will flush to disk. This \
+                 provides a middle ground between pure in-memory (default) and always-on-disk \
+                 (--enable-accounts-disk-index) modes. Cannot be used with \
+                 --enable-accounts-disk-index or --accounts-index-limit-mb.",
+            ),
+    )
+    .arg(
+        Arg::with_name("accounts_index_limit_mb")
+            .long("accounts-index-limit-mb")
+            .value_name("MEGABYTES")
```

Comment:
I think we should only allow specifying threshold in bytes, not num elements of capacity in the hashmaps.

I also think we should use bytes instead of megabytes.